lib/improver/wxcode/weather_symbols.py

```diff
@@ -83,16 +82,30 @@ def check_input_cubes(self, cubes):
             for diagnostic, threshold, condition in zip(
                     diagnostics, thresholds, conditions):
 
+                # First we check the diagnostic name and units, performing
+                # a conversion is required and possible.
+                test_condition = (iris.Constraint(name=diagnostic))
+                matched_cube = cubes.extract(test_condition)
+                if not matched_cube:
+                    missing_data.append([diagnostic, threshold, condition])
+                    continue
+                else:
+                    cube_threshold_units = (
+                        matched_cube[0].coord('threshold').units)
+                    threshold.convert_units(cube_threshold_units)
+
+                # Then we check if the required threshold is present in the
+                # cube, and that the thresholding is relative to it correctly.
                 threshold = threshold.points.item()
                 test_condition = (
-                    Constraint(
-                        name=diagnostic,
-                        coord_values={'threshold': threshold},
+                    iris.Constraint(
+                        threshold=lambda cell:
+                            threshold*0.99 < cell < threshold*1.01,
                         cube_func=lambda cube: (
                             cube.attributes['relative_to_threshold'] ==
                             condition)))
-
-                if not cubes.extract(test_condition):
+                matched_threshold = matched_cube.extract(test_condition)
```

Comment:
I'm not sure you have a test that actually tests this functionality. You have one that tests if a cube is missing, but not one if the threshold is missing. This should be quite easy to set up, by changing one of your input cubes by doing something like cube=cube[0] to remove some of the thresholds in your input cube.
 
There should also be a check where the thresholds are in the wrong units to show this works.

What do you think?

Reply:
I've added the former, the latter already exists (test_incorrect_units). I've also modified the tests to use the float_tolerance defined in the __init__ so if we change that the tests still pass.